app/src/org/odk/collect/android/widgets/GeoPointWidget.java

```diff
@@ -60,100 +59,86 @@ public GeoPointWidget(Context context, FormEntryPrompt prompt) {
-        mAppearance = prompt.getAppearanceHint();
+        String appearance = prompt.getAppearanceHint();
+        // use maps or not
+        if (appearance != null && appearance.equalsIgnoreCase("maps")) {
+            try {
+                // do google maps exist on the device
+                Class.forName("com.google.android.maps.MapActivity");
+                mUseMaps = true;
+            } catch (ClassNotFoundException e) {
+                mUseMaps = false;
+            }
+        }
 
         setOrientation(LinearLayout.VERTICAL);
 
-        TableLayout.LayoutParams params = new TableLayout.LayoutParams();
-        params.setMargins(7, 5, 7, 5);
-
-        mGetLocationButton = new Button(getContext());
-        mGetLocationButton.setPadding(20, 20, 20, 20);
-        mGetLocationButton.setText(StringUtils.getStringSpannableRobust(getContext(), R.string.get_location));
-        mGetLocationButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontsize);
-        mGetLocationButton.setEnabled(!prompt.isReadOnly());
-        mGetLocationButton.setLayoutParams(params);
-
-        // setup play button
-        mViewButton = new Button(getContext());
-        mViewButton.setText(StringUtils.getStringSpannableRobust(getContext(), R.string.show_location));
-        mViewButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontsize);
-        mViewButton.setPadding(20, 20, 20, 20);
-        mViewButton.setLayoutParams(params);
-
-        // on play, launch the appropriate viewer
-        mViewButton.setOnClickListener(new View.OnClickListener() {
-            /*
-             * (non-Javadoc)
-             * @see android.view.View.OnClickListener#onClick(android.view.View)
-             */
-            @Override
-            public void onClick(View v) {
-
-                String s = mStringAnswer.getText().toString();
-                String[] sa = s.split(" ");
-                double gp[] = new double[4];
-                gp[0] = Double.valueOf(sa[0]).doubleValue();
-                gp[1] = Double.valueOf(sa[1]).doubleValue();
-                gp[2] = Double.valueOf(sa[2]).doubleValue();
-                gp[3] = Double.valueOf(sa[3]).doubleValue();
-                Intent i = new Intent(getContext(), GeoPointMapActivity.class);
-                i.putExtra(LOCATION, gp);
-                ((Activity) getContext()).startActivity(i);
-
-            }
-        });
-
         mStringAnswer = new TextView(getContext());
-
         mAnswerDisplay = new TextView(getContext());
         mAnswerDisplay.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontsize);
         mAnswerDisplay.setGravity(Gravity.CENTER);
 
+        Spannable locButtonText;
+        boolean viewButtonEnabled;
+
         String s = prompt.getAnswerText();
-        if (s != null && !s.equals("")) {
-            mGetLocationButton.setText(StringUtils.getStringSpannableRobust(getContext(), R.string.replace_location));
+        if (s != null && !("".equals(s))) {
```

Comment:
In this case we still want to make sure that 's' isn't null. Doing just "".equals(s) doesn't achieve this.